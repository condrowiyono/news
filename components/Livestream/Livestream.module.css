.container {
  position: relative;
}

.imageContainer {
  width: 155px;
  height: 230px;
  border-radius: 8px;
  flex-shrink: 0;
  box-sizing: border-box;
}

.imageContainer:after {
  content:'';
  position:absolute;
  left:0;
  top:0;
  width:100%;
  height:100%;
  border-radius: 8px;
  background: linear-gradient(0deg, rgba(0, 0, 0, 0.859) 20.8%, rgba(68, 68, 68, 0.165) 88.19%, rgba(84, 84, 84, 0) 122.9%);
}

.timestamp {
  position: absolute;
  left: 12px;
  top: 12px;
  background: #ADB1B4;
  border-radius: 6px;
  font-size: 12px;
  padding: 2px 8px;
}

.timestamp.clip {
  font-weight: 700;
  color: #FFFFFF;
}

.timestamp.scheduled {
  font-weight: 400;
  background: #FBD2D4;
  color: #ED2227;
}

.timestamp.live {
  background: rgba(0, 0, 0, 0.3);
  backdrop-filter: blur(4px);
  border-radius: 15px;
  color: #FFFFFF;
  font-weight: 700;
  font-size: 10px;
}

.timestamp.live .liveText {
  display: inline-flex;
  align-items: center;
}

.liveIndicator {
  margin-left: 2px;
  color: #ED2227;
  animation: blinker 1s linear infinite;
}

.content {
  position: absolute;
  left: 12px;
  right: 12px;
  bottom: 12px;
}

.category {
  font-size: 11px;
  line-height: 13px;
  color: #FFFFFF;
}

.title {
  font-weight: 900;
  font-size: 14px;
  line-height: 17px;
  color: #FFFFFF;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;  
  overflow: hidden;
  margin-bottom: 4px;
}

@keyframes blinker {
  50% {
    opacity: 0;
  }
}