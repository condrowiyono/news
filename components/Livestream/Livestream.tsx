import style from "./Livestream.module.css";
import ImageServer from "@components/Image";
import Skeleton from "@components/Skeleton";
import { formatNumber } from "@utils/number";

export const SkeletonLivestream = () => {
  return (
    <div className={style.container} data-testid="title">
      <Skeleton height={230} width={155} />
      <div className={style.content}>
        <Skeleton height={24} width={120} />
        <Skeleton height={12} width={48} />
      </div>
    </div>
  );
};

const Livestream = ({
  title,
  category,
  live_event,
  image,
  date,
  viewer_count,
}: INews) => {
  return (
    <div className={style.container}>
      <div className={style.imageContainer}>
        <ImageServer
          src={image}
          objectFit="cover"
          width={155}
          height={230}
          alt="profile-picture"
          style={{ borderRadius: 8 }}
        />
      </div>
      {live_event ? (
        <div className={`${style.timestamp} ${style.live}`} data-testid="live">
          <div className={style.liveText}>
            <ImageServer
              src="/icons/ic_eye_open.png"
              width={10}
              height={10}
              alt="live"
              style={{ borderRadius: 8 }}
            />
            {formatNumber(viewer_count || 0, true)} |{" "}
            <span className={style.liveIndicator}> â€¢ Live</span>
          </div>
        </div>
      ) : (
        <div
          data-testid="not-live"
          className={`${style.timestamp} ${
            new Date() > new Date(date) ? style.scheduled : style.clip
          }`}
        >
          <div className={style.liveText}>
            {new Date() > new Date(date) ? "Terjadwal" : "Clip"}
          </div>
        </div>
      )}
      <div className={style.content}>
        <div className={style.title}>{title}</div>
        <div className={style.category}>{category}</div>
      </div>
    </div>
  );
};

export default Livestream;
