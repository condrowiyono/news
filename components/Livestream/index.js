import Livestream, { SkeletonLivestream } from "./Livestream";

export { SkeletonLivestream };
export default Livestream;