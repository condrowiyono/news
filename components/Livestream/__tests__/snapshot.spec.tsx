import { render } from "@testing-library/react";
import Livestream, { SkeletonLivestream } from "@components/Livestream";
import { live } from "./Livestream.spec";

describe("Livestream List", () => {
  it("renders Livestream unchanged", () => {
    const { container } = render(<Livestream {...live} />);
    expect(container).toMatchSnapshot();
  });

  it("renders SkeletonLivestream unchanged", () => {
    const { container } = render(<SkeletonLivestream />);
    expect(container).toMatchSnapshot();
  });
});
