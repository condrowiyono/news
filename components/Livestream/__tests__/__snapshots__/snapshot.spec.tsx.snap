// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Livestream List renders Livestream unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="imageContainer"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27155%27%20height=%27230%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="profile-picture"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="border-radius: 8px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/gif;base64,R0lGODlhAQABAPAAAL29vf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==);"
        />
        <noscript />
      </span>
    </div>
    <div
      class="timestamp live"
      data-testid="live"
    >
      <div
        class="liveText"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2710%27%20height=%2710%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="live"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="border-radius: 8px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/gif;base64,R0lGODlhAQABAPAAAL29vf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==);"
          />
          <noscript />
        </span>
        10.0rb
         |
         
        <span
          class="liveIndicator"
        >
           â€¢ Live
        </span>
      </div>
    </div>
    <div
      class="content"
    >
      <div
        class="title"
      >
        Judul
      </div>
      <div
        class="category"
      >
        Bisnis
      </div>
    </div>
  </div>
</div>
`;

exports[`Livestream List renders SkeletonLivestream unchanged 1`] = `
<div>
  <div
    class="container"
    data-testid="title"
  >
    <div
      class="skeleton"
      data-testid="skeleton"
      style="height: 230px; width: 155px; border-radius: 4px;"
    />
    <div
      class="content"
    >
      <div
        class="skeleton"
        data-testid="skeleton"
        style="height: 24px; width: 120px; border-radius: 4px;"
      />
      <div
        class="skeleton"
        data-testid="skeleton"
        style="height: 12px; width: 48px; border-radius: 4px;"
      />
    </div>
  </div>
</div>
`;
